### PR TITLE
Ecosystem-scale calculations and Fs calculation integrated

### PR DESCRIPTION
**Pull request by stephpenn1**

### CODE REVIEW

File: R/sapflow-Rs.Rmd [hunk 1]
```diff
@@ -218,24 +218,59 @@ ggplot(wx_dat, aes(DOY, PAR)) +
...
+```{r sapwood-area-calc, echo = FALSE}
+
+inventory %>% 
+  filter(Plot == "Control", Species_code != "FAGR") %>% 
+  drop_na(Sapflux) %>% 
+  select(Sapflux, DBH_2019, Species_code) %>% 
+  rename(Tree = Sapflux, DBH = DBH_2019, Species = Species_code) -> js_trees
+
+# To calculate sapflux, sapwood area needs to be calculated from sapwood depth
+sapwood_depth <- read_csv("~/Desktop/sapwood_depthsSERC.csv",
```
Comment (bpbond) [inline R/sapflow-Rs.Rmd:230]: This will break on my system–put file in repository and refer to it there.

File: R/sapflow-Rs.Rmd [hunk 2]
```diff
@@ -218,24 +218,59 @@ ggplot(wx_dat, aes(DOY, PAR)) +
...
+# Test to make sure predictor works 
+ggplot(sapwood_depth, aes(x = DBH, y = sapwood_d, colour = Species)) + 
+  geom_point() + 
+  geom_smooth(method = lm, se =  FALSE) + 
+  geom_point(data = js_trees, aes(x = DBH, y = pred_sd, 
+                                  colour = Species), shape = 1, size = 5)
+
+js_trees %>% 
+  # calculate area of the tree and heartwood, the difference will give us sapwood area
+  mutate(t_area_cm2 = pi * (DBH / 2)^2, hw_area_cm2 = pi * ((DBH - 2 * pred_sd) / 2) ^ 2) %>%
```
Comment (bpbond) [inline R/sapflow-Rs.Rmd:248]: Perhaps split to separate lines for clarity

File: R/sapflow-Rs.Rmd [hunk 3]
```diff
@@ -218,24 +218,59 @@ ggplot(wx_dat, aes(DOY, PAR)) +
...
 ```{r data-cleaning, echo = FALSE, message = FALSE}
 
 # First, we need to filter for the Js trees that also have Rs measurements
 rs_trees <- unique(ports_lt$Tree)
 
 sapflow_k_long %>% 
+  left_join(sapwood_area, by = "Tree") %>% 
   # Then, we need to transform K to Js (sapflux density) through a conversion
-  mutate(Js = 119* 10^(-6) * K^(1.231)) %>% 
+  mutate(Js = 119* 10^(-6) * K^(1.231)) %>%
```
Comment (bpbond) [inline R/sapflow-Rs.Rmd:261]: Where do the 119 and 1.231 come from? Is there a citation we can put here?

File: R/sapflow-Rs.Rmd [hunk 4]
```diff
@@ -218,24 +218,59 @@ ggplot(wx_dat, aes(DOY, PAR)) +
...
+  # Then sapflux density to sapflux (F, cm^3/s)
```
Comment (bpbond) [inline R/sapflow-Rs.Rmd:262]: Nice commenting 👏

**Review by bpbond — CHANGES_REQUESTED** [review]
A couple suggestions but very nice :)

File: R/sapflow-Rs.Rmd (hunk 1 above)
Comment (bpbond) [reply R/sapflow-Rs.Rmd:230]: This is the important one!

Comment (stephpenn1) [reply R/sapflow-Rs.Rmd:230]: meant to change that - ty for catching this!

File: R/sapflow-Rs.Rmd (hunk 3 above)
Comment (stephpenn1) [reply R/sapflow-Rs.Rmd]: This is from Granier 1987, 10.1093/treephys/3.4.309. I've added this in the comments

**Pull request merged**